### PR TITLE
Add preset support

### PR DESCRIPTION
**Pull request by DaanDeMeyer**
Presets can be defined in mkosi.presets/. A preset is just like a
regular config file/directory, except that mkosi can build multiple
presets sequentially.

If mkosi.presets/ exists, for each preset mkosi will read the global
configuration, followed by the individual preset configuration. It
will then build each of the presets in alpha-numerical order. Later
presets can use outputs of earlier presets, specifically using the
BaseTrees= and Initrds= options.

While this has many use cases, one promising use case is to allow
building an initrd and a final image that uses that initrd within a
single invocation of mkosi.

### CODE REVIEW

File: mkosi/config.py [hunk 1]
```diff
@@ -1697,23 +1710,51 @@ def parse(self, argv: Optional[Sequence[str]] = None) -> tuple[MkosiArgs, MkosiC
...
-            if s.default_factory:
-                default = s.default_factory(namespace)
-            elif s.default is None:
-                default = s.parse(s.dest, None, namespace)
-            else:
-                default = s.default
+                with chdir(p if p.is_dir() else Path.cwd()):
+                    self.parse_config(p if p.is_file() else Path("."), cp)
+
+                self.parse_config(Path("."), cp)
```
Comment (behrmann) [inline mkosi/config.py:1726]: I think this section could do with a few comments. The idea is that we pop into the profile dir, parse the profile config and then pop back and parse our global config there?

File: mkosi/config.py [hunk 2]
```diff
@@ -1697,23 +1710,51 @@ def parse(self, argv: Optional[Sequence[str]] = None) -> tuple[MkosiArgs, MkosiC
...
+
+                name = p.name.lstrip(string.digits + "-").removesuffix(".conf")
```
Comment (behrmann) [inline mkosi/config.py:1728]: This turns up a few times, let's factor it out
```python
def profilename_from_filename(name: str):= -> str:
    return name.lstrip(string.digits + "-").removesuffix(".conf")
```

File: mkosi/__init__.py [hunk 3]
```diff
@@ -1130,75 +1130,77 @@ def print_summary(args: MkosiArgs, config: MkosiConfig) -> None:
...
+
+    {bold("DISTRIBUTION")}:
+                Distribution: {bold(config.distribution.name)}
+                    Release: {bold(none_to_na(config.release))}
+                Architecture: {config.architecture}
+                        Mirror: {none_to_default(config.mirror)}
+        Local Mirror (build): {none_to_none(config.local_mirror)}
+    Repo Signature/Key check: {yes_no(config.repository_key_check)}
+                Repositories: {",".join(config.repositories)}
+                    Initrds: {",".join(os.fspath(p) for p in config.initrds)}
```
Comment (behrmann) [inline mkosi/__init__.py:1147]: The alignment is off here.

**Review by behrmann — COMMENTED** [review]
Really looking forward to this. :)

File: mkosi/config.py (hunk 2 above)
Comment (DaanDeMeyer) [reply mkosi/config.py]: This was accidentally duplicated, dropped

File: mkosi/__init__.py [hunk 4]
```diff
@@ -1574,6 +1567,14 @@ def run_build_script(state: MkosiState) -> None:
     if state.config.install_dir:
         empty_directory(state.config.install_dir)
 
+    # Create a few necessary mount points inside the build overlay.
```
[the remark this answers is not in the document]
Comment (DaanDeMeyer) [reply mkosi/__init__.py:1555]: With that change, we don't create the mount points anymore if there are no build packages (which is wrong, we still need to create them), so I moved the logic here to make sure the mountpoints are always created.

**Review by bluca — APPROVED** [review]
There's some static analysis warnings to fix or dismiss

**Pull request merged**